include/matx/kernels/resample_poly.cuh

```diff
@@ -40,24 +40,38 @@
 #include <stdio.h>
 #include <vector>
 
+#include <cooperative_groups.h>
+#include <cooperative_groups/reduce.h>
+
+namespace cg = cooperative_groups;
+
 #include "cuComplex.h"
 #include "matx/core/utils.h"
 #include "matx/core/type_utils.h"
 #include "matx/core/tensor_utils.h"
 
 namespace matx {
 
+// Use for __launch_bounds__ to allow the compiler to tune register usage
+static constexpr int MATX_RESAMPLE_POLY_MAX_NUM_THREADS = 256;
+
+// We use a static 11 KiB buffer to potentially store the filter. If it fits, we will load it
+// into smem. If not, then we will load it from global memory at the time of use. We choose
+// 11 KiB so that we can definitely fit four blocks in 48 KiB, leaving 1 KiB per block

```

Comment:
Have you experimented with dynamic shm and increasing this over L1 slicing?

Reply:
Do you mean adjusting the smem/L1 split via carveout? I have not tried increasing the shared memory allotments, but overall the improvements that I have seen by using shared memory have been relatively modest. If we have cases with relatively long filters (either due to large upsampling/downsampling factors, or due to strict filter performance requirements) then it's worth exploring. The kernel that is implemented but now unused decomposes the problem by phase, so only needs to store the taps for a single phase in smem. That would be ~1400 taps for FP64, so we may just be able to use that existing kernel in such cases without increasing the carveout.